torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -308,7 +308,11 @@ void Reducer::mark_variable_ready_dense(VariableIndex index) {
                       "bucket_view.sizes() = ", bucket_view.sizes(),
                       "bucket_view.strides() = ", bucket_view.strides());
     }
-    bucket_view.copy_(grad, /* non_blocking */ true);
+    // imitates wrapped_scalar_tensor in ATen/native/BinaryOps.cpp
+    auto wrapped = c10::scalar_to_tensor(float(1.0/process_group_->getSize()));
```

Comment:
Does the device of this scalar tensor matter? Does it need to stay on the same device as the grad/device?

Reply:
As long as the scalar is on CPU, it will be moved to the right GPU device automatically.

Reply:
CPU scalar is lambda-captured into the kernel https://github.com/pytorch/pytorch/blob/6d13b583a71dfb25640a1386e28526f82ad7c3bf/aten/src/ATen/native/cuda/Loops.cuh#L88-L104